src/data_source/storage/sql.rs

```diff
@@ -1789,6 +1800,534 @@ impl SqlStorage {
+                        histograms.block_size.push(size);
+                        histograms.block_transactions.push(num_transactions);
+                        histograms.block_heights.push(height);
+                        Ok(histograms)
+                    },
+                ).await;
+
+            histograms
+        }.await?;
+
+        let genesis_overview = async {
+            let row = self
+                .query_one(
+                    "SELECT
+                    (SELECT MAX(height) + 1 FROM header) AS blocks,
+                    (SELECT COUNT(*) FROM transaction) AS transactions",
+                    Vec::<Box<dyn ToSql + Send + Sync>>::new(),
+                )
+                .await?;
+
+            let blocks: i64 = row.try_get("blocks").map_err(|e| QueryError::Error {
+                message: format!("failed to get column blocks {e}"),
+            })?;
+            let transactions: i64 = row.try_get("transactions").map_err(|e| QueryError::Error {
+                message: format!("failed to get column transactions {e}"),
+            })?;
+
+            let blocks: u64 = blocks.try_into().map_err(|e| QueryError::Error {
+                message: format!("failed to convert blocks to u64 {e}"),
+            })?;
+            let transactions: u64 = transactions.try_into().map_err(|e| QueryError::Error {
+                message: format!("failed to convert transactions to u64 {e}"),
+            })?;
+
+            Ok::<_, QueryError>(GenesisOverview {
+                rollups: 0,
+                transactions,
+                blocks,
+            })
+        }
+        .await?;
+
+        let latest_block: BlockDetail<Types> =
+            self.get_block_detail(&BlockIdentifier::Latest).await?;
+        let latest_blocks: Vec<BlockSummary<Types>> = self
+            .get_block_summaries(&GetBlockSummariesRequest(BlockRange {
+                target: BlockIdentifier::Latest,
+                num_blocks: NonZeroUsize::new(10).unwrap(),
+            }))
+            .await?;
+        let latest_transactions: Vec<TransactionSummary<Types>> = self
+            .get_transaction_summaries(&GetTransactionSummariesRequest {
+                range: TransactionRange {
+                    target: TransactionIdentifier::Latest,
+                    num_transactions: NonZeroUsize::new(10).unwrap(),
+                },
+                filter: TransactionSummaryFilter::None,
+            })
+            .await?;
+
+        Ok(ExplorerSummary {
+            genesis_overview,
+            latest_block,
+            latest_transactions,
+            latest_blocks,
+            histograms,
+        })
+    }
+
+    async fn get_search_results(&self, query: String) -> QueryResult<SearchResult<Types>> {
+        let block_query = format!(
+            "SELECT {BLOCK_COLUMNS}
+                FROM header AS h
+                JOIN payload AS p ON h.height = p.height
+                WHERE h.hash LIKE $1 || '%'
```

Comment:
this would be an expensive query. Maybe we can use gist/gin index for this ? https://www.postgresql.org/docs/current/pgtrgm.html @jbearer

Reply:
Hm I thought btree index was applicable in the special case where you have `LIKE 'something%' (because that corresponds to lexicographic ordering).

Reply:
This query is for full text search or just searching the prefix?

Reply:
It should just be a prefix search I believe.

Usually when doing FTS you end up with a keyword like `MATCH` or something.  Though, I'm not as certain if this is true in `postgres`.

Reply:
Yeah, I was able to confirm that `Full Text Search` has its own functions and calling symantics, in addition to requiring specific index setup.  Apparently there are multiple versions, the one I read about specifically was `to_tsquery` for it's `MATCH` command.

the `||` in this case is the concatenation operator, though the `CONCAT` function exists, I believe in `postgres`, perhaps changing that could help prevent any confusion?